server/internal/infrastructure/db/postgres/market_hour_repo.go

```diff
@@ -0,0 +1,100 @@
+package pgdb
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/ark-network/ark/server/internal/core/domain"
+	"github.com/ark-network/ark/server/internal/infrastructure/db/postgres/sqlc/queries"
+)
+
+type marketHourRepository struct {
+	db      *sql.DB
+	querier *queries.Queries
+}
+
+func NewMarketHourRepository(config ...interface{}) (domain.MarketHourRepo, error) {
+	if len(config) != 1 {
+		return nil, fmt.Errorf("invalid config")
+	}
+	db, ok := config[0].(*sql.DB)
+	if !ok {
+		return nil, fmt.Errorf("cannot open market hour repository: invalid config, expected db at 0")
+	}
+
+	return &marketHourRepository{
+		db:      db,
+		querier: queries.New(db),
+	}, nil
+}
+
+func (r *marketHourRepository) Get(ctx context.Context) (*domain.MarketHour, error) {
+	marketHour, err := r.querier.GetLatestMarketHour(ctx)
+	if errors.Is(err, sql.ErrNoRows) {
+		return nil, nil
+	}
+	if err != nil {
+		return nil, fmt.Errorf("failed to get market hour: %w", err)
```

Comment:
either we replace the error or we return as it is, do not wrap it around a `failed to...` message
```suggestion
		return nil, err
```